### PR TITLE
feature: footer

### LINKED ISSUE #40: Footer
Refer the Figma design for the footer.

### PR DESCRIPTION
**Pull request by Kanishkk-1**
Fixes #40

### CODE REVIEW

File: app/footer.tsx
Comment (SaketRaman2002) [inline app/footer.tsx:11]: please use svg instead of png

**Comment by SaketRaman2002** [comment]
Please move all the images in a folder called assets and put the assets folder in components

File: app/footer.tsx
Comment (Kanishkk-1) [reply app/footer.tsx]: Done

**Comment by Kanishkk-1** [comment]
Resolved all the stated queries

File: app/page.tsx
Comment (GetPsyched) [inline app/page.tsx:6]: Uh?

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:83]: Make this into an object that is read from instead of manually initialising each list element

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:59]: It will be `text` and `link`. Style is common to all and won't be included in the object

**Comment by GetPsyched** [comment]
Typo earlier, move the assets to `public/assets/`. Also, rebase on master and format all these files.
```sh
git checkout master
git pull
git checkout feature/footer
git rebase master
```

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Typo earlier, move the assets to `public/assets/`. Also, rebase on master and format all these files.
```sh
git checkout master
git pull
git checkout feature/footer
git rebase master
```

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:164]: Please format and remove these issues
```suggestion
        <div className="flex flex-row items-center justify-center flex-shrink-0 py-5 lg:pr-10 sm:pr-4">
```

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:67]:
```suggestion

function footer() {
```

File: app/page.tsx
Comment (GetPsyched) [reply app/page.tsx:6]: Fix this

File: styles/globals.css
Comment (GetPsyched) [inline styles/globals.css:9]: formatting?? fix please.

File: tailwind.config.ts
Comment (GetPsyched) [inline tailwind.config.ts:19]: rebase onto master, this has been added already.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:1]:
```suggestion
```

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:3]:
```suggestion

import footerimg from "../public/assets/footerimg.png";
```

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: styles/globals.css
Comment (GetPsyched) [inline styles/globals.css:1]: Don't commit any changes in this file.

File: .vscode/settings.json
Comment (GetPsyched) [inline .vscode/settings.json:1]: remove this.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:67]:
```suggestion
export default function footer() {
```

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx]: bring this back.

File: app/page.tsx
Comment (GetPsyched) [inline app/page.tsx:4]: what, stop editing this file

File: tailwind.config.ts
Comment (GetPsyched) [reply tailwind.config.ts:19]: resolve this also

File: app/page.tsx
Comment (GetPsyched) [reply app/page.tsx:4]: this is not resolved, why marked as resolved. if you're confused, ask me, stop fighting with Git.

File: app/page.tsx
Comment (GetPsyched) [inline app/page.tsx:1]: I said stop editing this file. Why is this still in the PR??

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:101]: Add a good alt message, don't just put it there to remove the NextJS error

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:82]: This is not how you divide a text into multiple lines. Put it in a single div and apply a max width

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:97]: Use `ul` and `li` tags for this, not divs.

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:2]: Get these images from Figma instead of keeping it in the Git tree. See #45 for reference

File: app/page.tsx
Comment (GetPsyched) [reply app/page.tsx:1]: Is there any doubt? you've still not removed the diff of this file

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/page.tsx
Comment (Kanishkk-1) [reply app/page.tsx:1]: I'm sorry i'm not able to resolve this issue , could you please walk me through it ?

File: app/footer.tsx
Comment (Kanishkk-1) [reply app/footer.tsx:2]: Okay so i should remove the public/assets folder ? And you want me to add the images through the urls from figma?

Comment (GetPsyched) [reply app/footer.tsx:2]: Yes.

File: app/page.tsx
Comment (GetPsyched) [inline app/page.tsx:1]: Run `npm run format` or however else you want to run prettier. And then commit the changes in this file. It should go back to the way it was and won't be shown in the diff.

**Comment by Kanishkk-1** [comment]
is it okay now ? i ran npm run format on that page and its now formatted.

**Comment by Kanishkk-1** [comment]
and yeah its not showing the the diff too

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:65]:
```suggestion
export default function Footer() {
```

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:108]:
```suggestion
            alt="Design Glyph"
```

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Fonts need to be the default but that's already done in another PR so we'll wait until that's merged. Please do the following small changes for now

**Comment by Kanishkk-1** [comment]
Done. Anymore changes?

**Comment by GetPsyched** [comment]
None from your end, thanks!

This PR will now wait for #45 and also a formal list of what needs to be included in the footer.

**Review by GetPsyched — CHANGES_REQUESTED** [review]
For a learning experience, try replacing every single div element with a suitable semantic HTML tag. Feel free to refer the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element) for this.

Also, the footer bg is breaking after the rebase, please resolve.

File: app/directors-corner.tsx
Comment (GetPsyched) [inline app/directors-corner.tsx:3]: Don't remove this line

File: app/directors-corner.tsx
Comment (GetPsyched) [inline app/directors-corner.tsx:9]: divs aren't allowed in the repo. I'd say edit the horses running component to allow taking in breakpoints

File: app/directors-corner.tsx
Comment (GetPsyched) [inline app/directors-corner.tsx:25]: Breaking stuff into more divs is not how you make things responsive.

**Review by GetPsyched — COMMENTED** [review]
I'm gonna try my hand at this.

**Comment by Kanishkk-1** [comment]
i ran git pull -f  on my local branch feature/footer

**Comment by Kanishkk-1** [comment]
Are any more changes required ?

**Review by GetPsyched — CHANGES_REQUESTED** [review]
- Each column should stick to the top
- Responsiveness is not implemented properly. Use flex and justify

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:65]: Will not merge any div tags. Please refer https://developer.mozilla.org/en-US/docs/Web/HTML/Element and update to the most relevant semantic tags.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:7]: These objects will be defined inside the Footer function, not globally.

Also, compress all objects into one, like:
```
const items = {
    quickLinks = [ { text: "", link: "" ],
    about = [ { text: "", link: "" ],
    departments = [ { text: "", link: "" ],
}

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:42]: Huh? How is this a blockquote? Please read what a tag is supposed to represent:
> The <blockquote> [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) element indicates that the enclosed text is an extended quotation.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:53]: This is not what aside is for. Use `address`

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:57]: Okay just what. Did you even read the docs. Looks like you've just asked ChatGPT to translate this.

> The <object> [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML) element represents an external resource

"external resource"

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:83]: This isn't required. Just use the image directly

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:96]: If you bother to read the docs, `menu` is a semantic _replacement_ for `ul`. You've nested both here.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:143]: `span` tags aren't allowed either. This is also an incorrect usage of it.

File: app/footer.tsx
Comment (GetPsyched) [inline app/footer.tsx:154]: Where's the figure here?

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Please don't use ChatGPT to make the tags semantic. I could've done it myself, but I asked you to do it so you learn.

**Comment by Kanishkk-1** [comment]
?

**Comment by GetPsyched** [comment]
Am busy with mid semester exams ATM. Will review in some time. Meanwhile, please rebase on master to remove the merge conflicts.

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Needs rebase to remove conflicts.

**Review by GetPsyched — APPROVED** [review]
@Kanishkk-1 please see the [files changed](https://github.com/nitkkr-dev/nitkkr/pull/47/files) to see how you can better your code quality.

Pro tip: Design the entire HTML before doing any styling. It forces you to use a minimal amount of tags instead of adding more tags when fighting with styling.

**Pull request merged**
